src/test/java/networkbook/model/person/WebLinkTest.java

```diff
@@ -0,0 +1,74 @@
+package networkbook.model.person;
+
+import static networkbook.testutil.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+public class WebLinkTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new WebLink(null));
+    }
+
+    @Test
+    public void constructor_invalidLink_throwsIllegalArgumentException() {
+        String invalidLink = "";
+        assertThrows(IllegalArgumentException.class, () -> new Email(invalidLink));
+    }
+
+    @Test
+    public void isValidLink() {
+        // null email
+        assertThrows(NullPointerException.class, () -> WebLink.isValidLink(null));
+
+        // blank email
+        assertFalse(WebLink.isValidLink("")); // empty string
+        assertFalse(WebLink.isValidLink(" ")); // spaces only
+
+        // missing parts
+        assertFalse(WebLink.isValidLink("example.")); // missing local part
+        assertFalse(WebLink.isValidLink("peterjack@")); // missing domain label
+
+        // invalid parts
+        assertFalse(WebLink.isValidLink("peterjack@-.com")); // invalid domain name
+        assertFalse(WebLink.isValidLink("exam_ple.com")); // underscore in domain name
```

Comment:
Will the link work if the link includes a path name and if the underscore is there, e.g. [https://www.pythonanywhere.com/user/test_underscore/](https://www.pythonanywhere.com/user/test_underscore/) ?

Reply:
Previously no, but have amended for this to be the case now!